### PR TITLE
Bio1 2016 q3 primeconnections

### PR DESCRIPTION
**Pull request by AG-88301**

### CODE REVIEW

File: solutions/bio1/PrimeConnections/index.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      <STitle>Solution</STitle>
+      <SText>
+        To carry out a BFS, we should generate all the adjacent prime numbers for each prime number, 
+        and then search through them to find the solution.
+      </SText>
+      <SSubtitle>Generating Primes</SSubtitle>
+      <SText>
+        Initially, we need an efficient way to check if a number is prime.
+        To do this, we can use the Sieve of Eratosthenes algorithm 
+        (see <Link href="https://cp-algorithms.com/algebra/sieve-of-eratosthenes.html">here</Link> for more detail)
```
Comment (myst-6) [inline solutions/bio1/PrimeConnections/index.tsx:25]: more details*

File: solutions/bio1/PrimeConnections/index.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        to generate all prime numbers up to the specified limit.
+        This will create an array of booleans, where the index represents the number and the value represents whether the number is prime, 
+        allowing us to efficiently check if a number is prime in $\mathcal{brace("O")}(1)$ time.
+      </SText>
+      <SSubtitle>Finding the Shortest Path</SSubtitle>
+      <SText>
+        Now we can carry out a breadth first search from the starting node. 
+        In a breadth first search, we maintain a queue of nodes to visit and their distance from the start, as well as a set of visited nodes.
+        For each node, we can iterate through all the prime numebrs that are $2^k$ away from the current node for some value $k$,
```
Comment (myst-6) [inline solutions/bio1/PrimeConnections/index.tsx:34]: numbers* typo

Comment (myst-6) [reply solutions/bio1/PrimeConnections/index.tsx:34]: i think 'integer' or 'whole number' instead of 'value' would be better

**Review by myst-6 — CHANGES_REQUESTED** [review]
mostly good

File: solutions/bio1/PrimeConnections/index.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      <SSubtitle>Generating Primes</SSubtitle>
+      <SText>
+        Initially, we need an efficient way to check if a number is prime.
+        To do this, we can use the Sieve of Eratosthenes algorithm 
+        (see <Link href="https://cp-algorithms.com/algebra/sieve-of-eratosthenes.html">here</Link> for more detail)
+        to generate all prime numbers up to the specified limit.
+        To do this, we first initialise an array of booleans, where the index represents the number and the value represents whether the number is prime.
+        Then we can iterate through each number flagged <SCode>true</SCode> in the array 
+        (each prime number, $0$ and $1$ should be flagged as <SCode>false</SCode> as they are not prime) 
+        and flag all of it's multiples as <SCode>false</SCode>.
```
Comment (myst-6) [inline solutions/bio1/PrimeConnections/index.tsx:30]: oh  no no no

**Review by myst-6 — DISMISSED** [review]

File: solutions/bio1/PrimeConnections/index.tsx (hunk 3 above)
Comment (AG-88301) [reply solutions/bio1/PrimeConnections/index.tsx:30]: ohhh
ik what u mean

**Comment by AG-88301** [comment]
im dying

**Review by macaquedev — APPROVED** [review]

**Pull request merged**